compiler/rustc_lint/src/pass_by_value.rs

```diff
@@ -78,7 +77,7 @@ fn gen_args(cx: &LateContext<'_>, segment: &PathSegment<'_>) -> String {
                     .tcx
                     .sess
                     .source_map()
-                    .span_to_snippet(c.span())
+                    .span_to_snippet(c.as_unambig_ct().span())
```

Comment:
why don't we just make this type also have a span lol

Reply:
gg good point, I made `ConstArg`'s inherent methods generic over `P` since it was a trivial change